types/database.ts

```diff
@@ -0,0 +1,165 @@
+import type { ColumnType } from 'kysely';
+
+export type Generated<T> = T extends ColumnType<infer S, infer I, infer U>
+	? ColumnType<S, I | undefined, U>
+	: ColumnType<T, T | undefined, T>;
+
+export type Json = ColumnType<JsonValue, string, string>;
+
+export type JsonArray = JsonValue[];
+
+export type JsonObject = {
+	[K in string]?: JsonValue;
+};
+
+export type JsonPrimitive = boolean | null | number | string;
+
+export type JsonValue = JsonArray | JsonObject | JsonPrimitive;
+
+export type Numeric = ColumnType<string, string | number, string | number>;
+
+export type Timestamp = ColumnType<Date, Date | string, Date | string>;
+
+export interface Agencies {
+	id: Generated<string>;
+	name: string;
+	website: string;
+	phone: string;
+	bio: string | null;
+	address_line_1: string;
+	address_line_2: string | null;
+	city: string;
+	state: string;
+	country: string;
+	zip_code: string;
+	verified: Generated<boolean>;
+	employer_identification_number: string;
+	account_manager_id: string;
+	image_id: string;
+	created_at: Generated<Timestamp>;
+	updated_at: Timestamp | null;
+}
+
+export interface Children {
+	id: Generated<string>;
+	first_name: string;
+	last_name: string;
+	birth_year: number;
+	interest: string;
+	story: string;
+	image_id: string;
+	created_at: Generated<Timestamp>;
+}
+
+export interface CommunityPosts {
+	id: Generated<string>;
+	message: string;
+	agency_id: string;
+	image_id: string;
+	created_at: Generated<Timestamp>;
+}
+
+export interface Images {
+	id: Generated<string>;
+	url: string;
+	meta_data: Json | null;
+	created_at: Generated<Timestamp>;
+}
+
+export interface Items {
+	id: Generated<string>;
+	name: string;
+	price: Numeric;
+	link: string;
+	retailer: string;
+	product_id: string;
```

Comment:
@Enubia 
let's add color and size string fields too 
I thought about the variation details but for most cases, we just need color and size details for the order
no need to complicate it by making a meta_data json field

Reply:
> Let's add color and size string fields too

If we want to add something like this we should go with the `meta_data` field.
This way we are future-proof in case we want to add additional fields and don't need to write migrations for those cases.

Reply:
but why? it will be harder to separate string inputs, as you don't know what the type is supposed to be.
we do not need to accommodate anything more than the color and size
and it will stay the same in the future too
@nick-w-nick do you have an insight on this?

Reply:
@stacysealky The only issue is that once we add other retailers they may not have the same variation options. Color and size are definitely a majority of them, but I could see multiple other types being relevant in the future, such as style, pattern, version, etc. The reason a simple metadata field that allows for free-form input is more fitting for the time being is:
- We don't restrict the agency to using a specific set of field for variation details in case it doesn't fit the color/size type
- We are simply adding this field as a trial run for agencies, I believe we decided that even the metadata field wasn't the most optimal solution, but by adding it we gain an opportunity for gathering data on how the agencies use it and based on that data we can decide if we need to add standardized variation fields like color/size as you mentioned, or if we need to find a different way to allow them to specify the details

Remember, once we integrate with Amazon, it will automatically populate the item on the page once they add the link, and will show the selected variation before they create the wishcard so they can verify they selected the correct one before proceeding.

Basically, we don't want to add static fields that relate to specific variations, as we aren't even sure how/if the agency plans to supply those details during creation and it will be difficult to move from specific fields to a generic field rather than vice versa if we decide to do so.

Reply:
agreed, ok let's keep it as meta_data